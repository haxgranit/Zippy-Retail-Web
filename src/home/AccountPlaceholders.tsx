import React from 'react';
import { useAppSelector } from '../app/hooks';
import { selectUser } from '../features/user/userSlice';

export default function AccountPlaceholders() {
  const user = useAppSelector(selectUser);

  return (
    <div style={{
      alignItems: 'center',
      display: 'flex',
      flexDirection: 'column',
      gap: '20px',
    }}
    >
      {!user.personal && (
        <div style={{
          border: 'dashed',
          float: 'right',
          padding: '20px',
          textAlign: 'center',
          width: '200',
        }}
        >
          PLACEHOLDER: Option to sign up for a personal account
        </div>
      )}
      {!user.business && (
        <div style={{
          border: 'dashed',
          float: 'right',
          padding: '20px',
          textAlign: 'center',
          width: '200',
        }}
        >
          PLACEHOLDER: Option to sign up for a business account
        </div>
      )}
    </div>
  );
}
