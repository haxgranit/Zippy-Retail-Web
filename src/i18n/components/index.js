import LegalEnCAText from "./legal/LegalEnCAText";
import LegalEnUSText from "./legal/LegalEnUSText";
import LegalEsUSText from "./legal/LegalEsUSText";
import LegalFrCAText from "./legal/LegalFrCAText";

export {
    LegalEnCAText,
    LegalEnUSText,
    LegalEsUSText,
    LegalFrCAText
}
