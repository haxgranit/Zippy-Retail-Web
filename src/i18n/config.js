import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import { LegalEnCAText, LegalEnUSText, LegalEsUSText, LegalFrCAText } from './components';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: {
        ...require('./locales/en-CA/translations.json'),
        legal: {
          text: (
            <>
              <LegalEnCAText />
            </>
          )
        }
      }
    },
    'en-US': {
      translations: {
        ...require('./locales/en-US/translations.json'),
        legal: {
          text: (
            <>
              <LegalEnUSText />
            </>
          )
        }
      }
    },
    'es-US': {
      translations: {
        ...require('./locales/es-US/translations.json'),
        legal: {
          text: (
            <>
              <LegalEsUSText />
            </>
          )
        }
      }
    },
    'fr-CA': {
      translations: {
        ...require('./locales/fr-CA/translations.json'),
        legal: {
          text: (
            <>
              <LegalFrCAText />
            </>
          )
        }
      }
    },
  },
  returnObjects: false,
  returnedObjectHandler: (key, value, options) => value,
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;