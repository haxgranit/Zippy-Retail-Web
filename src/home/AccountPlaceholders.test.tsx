import '../i18n/config';
import AccountPlaceholders from './AccountPlaceholders';

const ReactTestRenderer = require('react-test-renderer');

describe('AccountPlaceholders Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(<AccountPlaceholders />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
