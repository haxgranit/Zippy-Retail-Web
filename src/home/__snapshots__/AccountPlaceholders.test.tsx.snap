// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPlaceholders Component matches the snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "gap": "20px",
      "width": "240px",
    }
  }
>
  <div
    style={
      Object {
        "border": "dashed",
        "float": "right",
        "padding": "20px",
        "textAlign": "center",
        "width": "200",
      }
    }
  >
    PLACEHOLDER: Option to sign up for a personal account
  </div>
  <div
    style={
      Object {
        "border": "dashed",
        "float": "right",
        "padding": "20px",
        "textAlign": "center",
        "width": "200",
      }
    }
  >
    PLACEHOLDER: Option to sign up for a business account
  </div>
</div>
`;
